derive_builder_core/src/macro_options/darling_opts.rs

```diff
@@ -241,6 +247,80 @@ pub struct Field {
     try_setter: Flag,
     #[darling(default)]
     field: FieldMeta,
+    #[darling(skip)]
+    field_attrs: Vec<Attribute>,
+    #[darling(skip)]
+    setter_attrs: Vec<Attribute>,
+}
+
+impl Field {
+    /// Remove the `builder_field_attrs(...)` packaging around attributes meant for fields
+    /// in the builder.
+    fn unnest_attrs(mut self) -> darling::Result<Self> {
+        let mut errors = vec![];
+
+        for attr in self.attrs.drain(..) {
+            let unnest = {
+                if attr.path.is_ident("builder_field_attrs") {
+                    Some(&mut self.field_attrs)
+                } else if attr.path.is_ident("builder_setter_attrs") {
+                    Some(&mut self.setter_attrs)
+                } else {
+                    None
+                }
+            };
+            if let Some(unnest) = unnest {
+                match unnest_from_one_attribute(attr) {
+                    Ok(n) => unnest.extend(n),
+                    Err(e) => errors.push(e),
+                }
+            } else {
+                self.field_attrs.push(attr.clone());
+                self.setter_attrs.push(attr);
+            }
+        }
+
+        if !errors.is_empty() {
+            return Err(darling::Error::multiple(errors));
+        }
+
+        Ok(self)
+    }
+}
+
+fn unnest_from_one_attribute(attr: syn::Attribute) -> darling::Result<Vec<Attribute>> {
+    match &attr.style {
+        syn::AttrStyle::Outer => (),
+        syn::AttrStyle::Inner(bang) => {
+            return Err(darling::Error::unsupported_format(
+                "builder_field_attrs must be an outer attributes",
```

Comment:
Nit: This function is called for `builder_field_attr` and `builder_setter_attr`; maybe get the path? Not a big deal - I'm not even sure how you'd legitimately get an inner attribute in this situation.

Reply:
Good catch on the message, but I think as you say this error can't happen.  I have made it more general in the text, rather than including the path, and added an in-code comment.  (Will push shortly.)